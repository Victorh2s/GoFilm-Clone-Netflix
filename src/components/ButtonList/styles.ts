import styled, { css } from 'styled-components';

export const button2 = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 10%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};


  :hover{
    background: rgba(255,255,255, 20%);
  }

  `}
`;

export const link2 = styled.a`
  ${({ theme }) => css`
  display: flex;
  text-decoration: none;
  justify-content: center;
  align-items: center;

>svg{
  width: 4rem;
  color: ${theme.colors.white};
  padding: 0 0.5rem;
}


  `}
`;
