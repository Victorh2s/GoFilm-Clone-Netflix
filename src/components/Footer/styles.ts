import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.dark};
    height: 10rem;

  `}
`;
export const Container = styled.section`
  ${({ theme }) => css`

    display: flex;
    text-align: center;
    justify-content: center;
    padding: ${theme.spacings.small};


  `}
`;

export const link = styled.a`
  ${({ theme }) => css`

  padding:${theme.spacings.small};

  >svg{
    width: 3rem;
    height: 4rem;
    color: ${theme.colors.white};
  }

  `}
`;
