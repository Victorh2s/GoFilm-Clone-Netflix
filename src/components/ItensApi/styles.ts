import styled, { css } from 'styled-components';

export const itens = styled.div`
  ${({ theme }) => css`

  font-size: 1.8rem;
  font-weight: bold;
  margin-top: 1.5rem;
  color: ${theme.colors.white};

  `}
`;

export const Vote = styled.div`
  ${({ theme }) => css`
  display: inline-block;
  color: ${theme.colors.green};
  margin-right: 1.5rem;


  `}
`;

export const Year = styled.div`
  display: inline-block;
  margin-right: 1.5rem;
`;

export const Season = styled.div`
  display: inline-block;
`;

export const Description = styled.div`
  ${({ theme }) => css`

  margin-top: 1.5rem;
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.mediumGray};
  max-width: 35%;


  `}
`;

export const Genres = styled.div`
  ${({ theme }) => css`

  margin-top: 1.3rem;
  color: ${theme.colors.mediumGray};
  font-size: ${theme.font.sizes.small};
  font-style: italic;

  `}
`;
