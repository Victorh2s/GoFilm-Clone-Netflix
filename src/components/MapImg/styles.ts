import styled from 'styled-components';

export const MovieFilm = styled.div`
  margin-left: 1rem;

`;

export const Link = styled.a`

`;

export const Image = styled.img`
  height: 25rem;
  width: 19rem;
  transform: scale(0.9);
  transition: all 200ms ease-in-out;
  cursor: pointer;
  :hover{
    transform: scale(1);
  }


`;
