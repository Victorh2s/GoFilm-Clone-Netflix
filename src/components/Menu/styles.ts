import styled, { css } from 'styled-components';

export const Menu = styled.menu`
  ${({ theme }) => css`

  display: flex;
  height: 6rem;
  top: 0;
  left:0;
  right: 0;
  background: transparent;
  align-items: center;
  justify-content: space-between;
  overflow: hidden;
  position: fixed;
  transition: all 200ms ease-in-out;
  z-index: 4;



    `}
`;

export const Logo = styled.a`
  padding: 5rem;
`;

export const Netflix = styled.img`
  ${({ theme }) => css`
  width: 12rem;
  height: 12rem;
  `}
`;

export const Avatar = styled.a`
  padding: 5rem;
`;

export const icon = styled.img`
  ${({ theme }) => css`
  width: 3rem;
  height: 3rem;
  `}
`;
