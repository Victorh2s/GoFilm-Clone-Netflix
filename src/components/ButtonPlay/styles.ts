import { Wrapper as Text } from '../Text/styles';
import styled, { css } from 'styled-components';

export const button = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 70%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};

  :hover{
    background: rgba(255,255,255, 80%);
  }

  `}
`;

export const link = styled.a`
  ${({ theme }) => css`
  text-decoration: none;
  justify-content: center;
  align-items: center;
  display: flex;

  ${Text}{
    color: ${theme.colors.dark};
  }

>svg{
  width: 4rem;
  color: black;
  padding: 0 0.5rem;
}
text-decoration: none;
color: ${theme.colors.dark};

  `}
`;
