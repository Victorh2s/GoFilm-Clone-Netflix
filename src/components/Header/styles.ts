import styled from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Wrapper = styled.section`
    height: 97vh;
`;

export const GladientVertical = styled.div`
width: inherit;
height:inherit;
background: linear-gradient(to top, #0d0c0c 5%, transparent 100%);

`;

export const GladientHorizontal = styled.div`

width: inherit;
height:inherit;
background: linear-gradient(to right , #0d0c0c 18%, transparent 95%);
display: flex;
flex-direction: column;
justify-content: center;
padding-left: 3rem;
padding-top: 4rem;

${Heading}{
  max-width: 50%;
}

`;
