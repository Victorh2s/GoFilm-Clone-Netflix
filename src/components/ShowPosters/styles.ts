import { Title as Heading } from '../Heading/styles';
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  ${Heading}{
    text-align: center;
    margin: 0 0 0.5rem 0;
  }
  padding:  0 0 5.5rem 0 ;
  background: ${theme.colors.dark};


  `}
`;

export const ButtonLeft = styled.button`
  ${({ theme }) => css`
  position: absolute;
  height: 27.1rem;
  z-index: 1;
  left: 0;
  background: linear-gradient(to right , black 15%, transparent);
  border: none;
  opacity: 0;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  >svg{
    width: 6rem;
    color: ${theme.colors.white};
  }

  `}
`;
export const ButtonRight = styled.button`
  ${({ theme }) => css`
  position: absolute;
  height: 27.1rem;
  display: flex;
  z-index: 1;
  right: 0;
  background: linear-gradient(to left , black 15%, transparent);
  border: none;
  opacity: 0;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  >svg{
    width: 6rem;
    color: ${theme.colors.white};
  }

  `}
`;

export const Container = styled.div`
  display: flex;
  transition: all 250ms ease-in-out;
  overflow:hidden;

  :hover{

    ${ButtonLeft}{
      opacity:1;
    }

    ${ButtonRight}{
      opacity:1;
    }
  }

`;
