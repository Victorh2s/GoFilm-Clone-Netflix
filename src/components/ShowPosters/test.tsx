import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ShowPosters, ShowPostersProps } from '.';
import { Science_fiction } from '../../service';

const props: ShowPostersProps = {
  APIurl: Science_fiction,
};

//heading
//button left
//button right

describe('<ShowPosters />', () => {
  it('should render heading', () => {
    renderTheme(<ShowPosters {...props} />);

    expect(
      screen.getByRole('heading', { name: 'Ficção Científica' }),
    ).toBeInTheDocument();
  });

  it('should put -150px in the left margin when right clicking', () => {
    renderTheme(<ShowPosters {...props} />);
    const button = screen.getByLabelText('Button Right');
    const Container = screen.getByLabelText('Container');

    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(Container).toHaveStyle({ 'margin-left': '-180px' });
  });

  it('should leave margin left at 0 when left clicking', () => {
    renderTheme(<ShowPosters {...props} />);
    const button = screen.getByLabelText('Button Left');
    const Container = screen.getByLabelText('Container');

    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(Container).toHaveStyle({ 'margin-left': '0px' });
  });

  it('should match to snapshot', () => {
    const { container } = renderTheme(<ShowPosters {...props} />);

    expect(container).toMatchSnapshot();
  });
});
