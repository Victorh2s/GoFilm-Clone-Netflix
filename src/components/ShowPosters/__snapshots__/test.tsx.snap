// Vitest Snapshot v1

exports[`<ShowPosters /> > should match to snapshot 1`] = `
.c2 {
  color: #FFFFFF;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 {
  padding: 0 0 5.5rem 0;
  background: #0d0c0c;
}

.c0 .c1 {
  text-align: center;
  margin: 0 0 0.5rem 0;
}

.c8 {
  position: absolute;
  height: 27.1rem;
  z-index: 1;
  left: 0;
  background: linear-gradient(to right ,black 15%,transparent);
  border: none;
  opacity: 0;
  cursor: pointer;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c8 >svg {
  width: 6rem;
  color: #FFFFFF;
}

.c5 {
  position: absolute;
  height: 27.1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 1;
  right: 0;
  background: linear-gradient(to left ,black 15%,transparent);
  border: none;
  opacity: 0;
  cursor: pointer;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c5 >svg {
  width: 6rem;
  color: #FFFFFF;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  overflow: hidden;
}

.c3:hover .c7 {
  opacity: 1;
}

.c3:hover .c4 {
  opacity: 1;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1 c2"
    >
      Ficção Científica
    </h2>
    <div
      aria-label="Container"
      class="c3"
      style="margin-left: 0px;"
    >
      <button
        aria-label="Button Right"
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="c6"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.776 1.553a.5.5 0 0 1 .671.223l3 6a.5.5 0 0 1 0 .448l-3 6a.5.5 0 1 1-.894-.448L9.44 8 6.553 2.224a.5.5 0 0 1 .223-.671z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <button
        aria-label="Button Left"
        class="c7 c8"
      >
        <svg
          aria-hidden="true"
          class="c6"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.224 1.553a.5.5 0 0 1 .223.67L6.56 8l2.888 5.776a.5.5 0 1 1-.894.448l-3-6a.5.5 0 0 1 0-.448l3-6a.5.5 0 0 1 .67-.223z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
